### PR TITLE
Fix hotdata method access; 

### PR DESCRIPTION
**Pull request by dannon**
this still needs a lot of love/refactoring (in general, after this PR) but this makes it work again, I think, and without a potential caching issue?  (opening to rerun selenium tests...)

xref #7251

### CODE REVIEW

**Comment by Nerdinacan** [comment]
That's basically what I did. There's a place in the template you're going to have to update as well

```
<hot-table
                        id="hot-table"
                        ref="hotTable"
                        :data="hotData['data']"
                        :colHeaders="colHeadersDisplay"
                        :readOnly="true"
                        stretchH="all"
                    >
```

**Comment by Nerdinacan** [comment]
So I took a careful look at this, and there was indeed a side-effect in the hotData computed, but it was completely unnecessary. Some processing was being bundled in an imported module in a function applyRules which mutated a passed array, triggering the eslint warning.

Looking carefully at what happens in that applyRules function, there was no real reason for it to be mutating the passed parameter, so I just made it an immutable input and generated a new computed value from the result of that processing and I think we're good.

File: client/galaxy/scripts/components/RuleCollectionBuilder.vue [hunk 1]
```diff
@@ -1029,7 +1029,7 @@ export default {
         }
         return {
             rules: rules,
-            colHeadersPerRule: [],
+            // colHeadersPerRule: [],
```
Comment (guerler) [inline client/galaxy/scripts/components/RuleCollectionBuilder.vue]: We probably should just remove this line.

Comment (Nerdinacan) [reply client/galaxy/scripts/components/RuleCollectionBuilder.vue]: Yup

**Comment by dannon** [comment]
@Nerdinacan Thanks for taking a look; I hadn't had a chance to dig deeper and figure out if the side effect (resetting `this.colHeadersPerRule` on access) was intentional and necessary or not.  Good to know it's not!

**Comment by Nerdinacan** [comment]
@dannon Yeah I got all the tests passing locally, but the entire selenium suite looks like it's not running at all. Not sure why yet.

**Comment by mvdbeek** [comment]
I've fixed the test setup in https://github.com/galaxyproject/galaxy/pull/7278, problem was an incompatible version of node in the client builder image.

**Review by nsoranzo — APPROVED** [review]
With this PR, Selenium test errors go down from 26 to 13, with no new error.

**Comment by nsoranzo** [comment]
The failed API test should be fixed by https://github.com/galaxyproject/galaxy/pull/7279 .

**Comment by dannon** [comment]
I'm checking the remaining failures manually and they're all passing on my local machine, looking for other configuration differences that'd explain it.

In any event, this is ready to merge.

**Comment by mvdbeek** [comment]
I see the same errors locally, this one for instance is also in the selenium logs:
screen shot 2019-01-31 at 15 39 21

**Comment by dannon** [comment]
@mvdbeek The last 13?  Yeah, I reproduced them yesterday in a virtual machine, working on fixes this morning.  Out of curiosity, what's your environment that reproduced them?  I couldn't get it to happen on my macbook.

**Comment by mvdbeek** [comment]
So weird, I tried the same thing just now again and it worked without error.

**Comment by dannon** [comment]
@mvdbeek I think it's a render timing thing, and fast machines won't reproduce it.  If I bump up the waits, tests pass, but I don't really like doing that so I'm trying a few more things.

**Pull request merged**